master/internal/command/tensorboard_manager.go

```diff
@@ -96,11 +104,63 @@ func (t *tensorboardManager) Receive(ctx *actor.Context) error {
 		}
 
 		actors.NotifyAfter(ctx, tickInterval, tensorboardTick{})
+	case TensorboardRequestWithUser:
+		summaryResp, err := t.handleTensorboardRequest(ctx, msg.User, &msg.Tensorboard)
+		if err != nil {
+			return errors.Wrap(err, "failed to launch tensorboard")
+		}
+		ctx.Respond(summaryResp.Get().(summary).ID)
 	}
 
 	return nil
 }
 
+func (t *tensorboardManager) handleTensorboardRequest(
+	ctx *actor.Context,
+	user *model.User,
+	req *TensorboardRequest,
+) (actor.Response, error) {
+	fmt.Println(req)
+	commandReq, err := parseCommandRequestWithUser(
+		*user, t.db, &req.commandParams, &t.taskSpec.TaskContainerDefaults)
+	if err != nil {
+		respondBadRequest(ctx, err)
+		return nil, err
+	}
+
+	if commandReq.AgentUserGroup == nil {
+		commandReq.AgentUserGroup = &t.defaultAgentUserGroup
+	}
+
+	if len(req.ExperimentIDs) == 0 && len(req.TrialIDs) == 0 {
+		respondBadRequest(ctx, errors.New("must set experiment or trial ids"))
+		return nil, err
+	}
+
+	ctx.Log().Infof("creating tensorboard (experiment id(s): %v trial id(s): %v)",
+		req.ExperimentIDs, req.TrialIDs)
+
+	b, err := t.newTensorBoard(commandReq, *req)
+
+	if err != nil {
+		ctx.Respond(err)
```

Comment:
I know it was like this before, but maybe we could wrap these errors?

Reply:
just to confirm: the errors returned from newTensorBoard already are descriptive but we want another layer on top to make it easier to see what happened?

Reply:
if they're descriptive enough I suppose it can be skipped.